contracts/wrapper/NameWrapper.sol

```diff
@@ -432,29 +454,37 @@ contract NameWrapper is
     ) public returns (uint64) {
         bytes32 node = _makeNode(parentNode, labelhash);
 
-        // this flag is used later, when checking fuses
-        bool canModifyParentName = canModifyName(parentNode, msg.sender);
-        // only allow the owner of the name or owner of the parent name
-        if (!canModifyParentName && !canModifyName(node, msg.sender)) {
-            revert Unauthorised(node, msg.sender);
-        }
-
         (address owner, uint32 fuses, uint64 oldExpiry) = getData(
             uint256(node)
         );
 
-        // Either CAN_EXTEND_EXPIRY must be set, or the caller must have permission to modify the parent name
-        if (!canModifyParentName && fuses & CAN_EXTEND_EXPIRY == 0) {
-            revert OperationProhibited(node);
+        (address parentOwner, uint32 parentFuses, uint64 parentExpiry) = getData(uint256(parentNode));
+
+        // Set the expiry between the old expiry and the parent expiry (max expiry). 
+        expiry = _normaliseExpiry(expiry, oldExpiry, parentExpiry);
+
+        // Allow the owner of the parent name to extend the expiry.
+        if(canModifyNameWithData(parentOwner, parentFuses, parentExpiry, msg.sender)){
+            _setData(node, owner, fuses, expiry);
+            emit ExpiryExtended(node, expiry);
+            return expiry;
+        } 
+
+        // If the caller is not the parent then check to make sure the caller is the owner of the name.
+        if (!canModifyNameWithData(owner, fuses, oldExpiry, msg.sender)) {
```

Comment:
Can you make a note in the comments that this needs to be oldExpiry. If we accidentally put expiry here in a future update, we might accidentally let names renew when they are already expired.

Reply:
Ok

Reply:
hmm, not sure I fully get this one. Can you write the comment and then I can just insert it? @jefflau